website/content/en/docs/ansible/development-tips.md

```diff
@@ -432,55 +461,54 @@ Declaring collections in the role meta allows you to invoke the
 ```
 
 ### Ansible Operator Conditions
-The Ansible Operator has a set of conditions which it will use as it performs
-its reconciliation procedure. There are only a few main conditions:
 
-* Running - the Ansible Operator is currently running the Ansible for
-  reconciliation.
+An Ansible Operator has a set of conditions that are used during reconciliation.
+There are mainly two condition types:
 
-* Successful - if the run has finished and there were no errors, the Ansible
-  Operator will be marked as Successful. It will then wait for the next
-  reconciliation action, either the reconcile period, dependent watches triggers
-  or the resource is updated.
+* `Running` - the Operator is currently running Ansible for reconciliation or
+  if a run has finished with no errors and it is waiting for the next
+  reconciliation action. The next action can be triggered by the reconcile
+  period, dependent watches or an update to the Custome Resource.
```

Comment:
The : 
```
or
  if a run has finished with no errors and it is waiting for the next
  reconciliation action. The next action can be triggered by the reconcile
  period, dependent watches or an update to the Custome Resource.
```
is not valid for `Running`. 

When the reconciliation finishes it should assume the `Successful` status.

Reply:
See the code source. The changes made here regards the status are not accurate. 

https://github.com/operator-framework/operator-sdk/blob/ad1d4d1953abf22c336f67198fc23a489c0f4fdb/pkg/ansible/controller/status/utils.go#L22-L30

Reply:
Well, I don't agree.
The only place the SuccessfulReason is used (referenced) is here - https://github.com/operator-framework/operator-sdk/blob/master/pkg/ansible/controller/reconcile.go#L386-L392. In that context, the condition *type* is still "Running". 

In fact, when we are talking about condition type, rather than the reason why the operator is in a specific condition type, there are only two choices, not three (https://github.com/operator-framework/operator-sdk/blob/master/pkg/ansible/controller/status/types.go#L88-L96)

```go
// ConditionType - type of condition
type ConditionType string

const (
	// RunningConditionType - condition type of running.
	RunningConditionType ConditionType = "Running"
	// FailureConditionType - condition type of failure.
	FailureConditionType ConditionType = "Failure"
)
```

Reply:
Hi @tenqm,  

If you check the [Memcached Sample](https://github.com/operator-framework/operator-sdk-samples/tree/master/ansible/memcached-operator) when the task finishes the Status will be : 

```
Status:
  Conditions:
    Ansible Result:
      Changed:             0
      Completion:          2020-07-04T09:41:01.05909
      Failures:            0
      Ok:                  2
      Skipped:             0
    Last Transition Time:  2020-07-04T09:40:42Z
    Message:               Awaiting next reconciliation
    Reason:                Successful
    Status:                True
    Type:                  Running
Events:                    <none>
```

See that the type of the conditional is `Running` but the Reason is `Successful` and the doc here is describing the status conditionals reason `Successful` and `Failed`. Because of this, could you please revert your changes and keep it as it was before?

Reply:
Why revert this? The original text lists three values: `Running`, `Successful` and `Failed`.
The set of condition types:

- `Running`
- `Failure`

The set of condition reasons:

- `Running`: valid when type is `Running`; 
- `Successful`: valid when type is `Running`;
- `Failed`: valid only when type is `Failure`;
- `Unknown`: defined, but not used anywhere, designed for `Failure` condition type.

Reply:
> Why revert this? The original text lists three values: Running, Successful and Failed.
Because the previous text was describing the `condition reasons` and you are changing the text. 

I agree that it has a hall for improvements. However, WDYT about we revert it and the working on this area in another PR just to improve the explanation over the status conditions and its reasons to make easier the review and etc. Ideally, each pr should have a close/small scope instead with fewer changes in order to ensure the quality and make the process flow smoothly.

Reply:
When you examine the status of a resource in Kubernetes, you will first look at the condition.type and then the condition.status. The combination of "type" and "status" gives you an idea whether the resource is operational. If you got an negative combination of the "type" and "status", you will check the "message" and "reason" field. In other words, we are not checking the reasons directly.

If we check the condition type definition again, I'm even thinking whether we should remove the `Failure` condition type because it is redundant. A `Failure` type of condition is equivalent to `Running=False`.

| type | status | reason | note |
|------|-------|--------|------|
| Running | True | Running | all good |
| Running | True | Successful | last reconciliation was successful |
| Running | False | Failed | a known error is caught |
| Running | False | Unknown | something bad happened with unknown reason |
| Running | Unknown | ... | ... |
| Failure | True | ... | ... |
| Failure | False | ... | ... |
| Failure | Unknown | ... | ... |

Anyway, I agree we can leave this untouched at the moment.